core/lib/spree/core/search/base.rb

```diff
@@ -89,7 +89,11 @@ def prepare(params)
 
             per_page = params[:per_page].to_i
             @properties[:per_page] = per_page > 0 ? per_page : Spree::Config[:products_per_page]
-            @properties[:page] = (params[:page].to_i <= 0) ? 1 : params[:page].to_i
+            if !params[:page].respond_to?(:to_i)
```

Comment:
I'd rather the results of this condition were switched so it just read: `params[:page].respond_to?(:to_i)`